client.go

```diff
@@ -116,7 +116,11 @@ func (c *Client) TimerAggregated(name string, value float64, tags Tags, aggregat
 	c.PutAggregated(name, value, tags, time.Now().Unix(), aggregation, frequency)
 }
 
-func (c *Client) Put(name string, value float64, tags Tags, timestamp int64, aggs Aggregations, freq AggregationFrequency) error {
+func (c *Client) Put(name string, value float64, tags Tags, timestamp int64, aggs Aggregations, freq AggregationFrequency, opts ...PutOption) error {
+	if len(opts) > 0 {
+		return c.buffer.Put(name, value, tags.Merge(c.globalTags), timestamp, aggs, freq, opts[0])
+	}
+
 	return c.buffer.Put(name, value, tags.Merge(c.globalTags), timestamp, aggs, freq)
 }
```

Comment:
This isn't correct, the client should just propagate the options to the buffer method - otherwise why would you have built the concept of option to support multiple options?

```suggestion
func (c *Client) Put(name string, value float64, tags Tags, timestamp int64, aggs Aggregations, freq AggregationFrequency, opts ...PutOption) error {
	return c.buffer.Put(name, value, tags.Merge(c.globalTags), timestamp, aggs, freq, opts...)
}
```

See it in action: https://play.golang.org/p/_AiTll8UppY